File: Dataset/Leetcode/test/100/453.js
 var XXX = function(p, q) {
    if (p === null && q === null) return true;
    if (p === null || q === null) return false;
    if (p.val === q.val && XXX(p.left, q.left) && XXX(p.right, q.right)) return true
    return false;
};

File: Dataset/Leetcode/train/111/453.java
class Solution {
    public int XXX(TreeNode root) {
        if (root == null) return 0;
        Queue<TreeNode> queue = new LinkedList<>();
        int depth = 0;
        queue.add(root);
        while(!queue.isEmpty()) {
            int size = queue.size();
            depth++;
            for (int i = 0; i < size; i++) {
                TreeNode curr = queue.poll();
                if (curr.left == null && curr.right == null) return depth;

                if (curr.left != null){
                    queue.add(curr.left);
                }
                if (curr.right != null) {
                    queue.add(curr.right);
                }
            }
        }
        return depth;

    }
}

